trade-finance-manager-api/src/v1/controllers/user/user.controller.upsert-tfm-user-from-entra-id-user.test.ts

```diff
@@ -0,0 +1,48 @@
+import { when, resetAllWhenMocks } from 'jest-when';
+import { aEntraIdUser } from '@ukef/dtfs2-common';
+import { generateSystemAuditDetails } from '@ukef/dtfs2-common/change-stream';
+import { mapUserData } from './helpers/mapUserData.helper';
+import { UserService } from '../../services/user.service';
+import { upsertTfmUserFromEntraIdUser } from './user.controller';
+import { userServiceMockResponses } from '../../../../test-helpers';
+
+jest.mock('../../services/user.service', () => ({
+  __esModule: true,
+  UserService: {
+    upsertTfmUserFromEntraIdUser: jest.fn(),
+  },
+}));
+
+describe('user controller', () => {
+  describe('upsertTfmUserFromEntraIdUser', () => {
+    const entraIdUser = aEntraIdUser();
+    const auditDetails = generateSystemAuditDetails();
+
+    const upsertedUserResponse = userServiceMockResponses.aUpsertTfmUserFromEntraIdUserResponse();
+    const mappedUserDetails = mapUserData(upsertedUserResponse);
+
+    beforeEach(() => {
+      jest.resetAllMocks();
+      resetAllWhenMocks();
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      when(UserService.upsertTfmUserFromEntraIdUser).calledWith({ entraIdUser, auditDetails }).mockResolvedValue(upsertedUserResponse);
+    });
+
+    it('should upsert user in the database', async () => {
+      await makeRequest();
+
+      // eslint-disable-next-line @typescript-eslint/unbound-method
+      expect(UserService.upsertTfmUserFromEntraIdUser).toHaveBeenCalledWith({ entraIdUser, auditDetails });
+    });
+
+    it('should return the mapped user', async () => {
+      const result = await makeRequest();
+
+      expect(result).toEqual(mappedUserDetails);
+    });
+
+    async function makeRequest() {
```

Comment:
Do we really need this, we could just have `await upsertTfmUserFromEntraIdUser({ entraIdUser, auditDetails });` in the `beforeEach`

Reply:
We need this -- We're looking at the result in the second test :)

Reply:
I think just having `await upsertTfmUserFromEntraIdUser({ entraIdUser, auditDetails });`  in individual tests would be nicer, but i'm not too fussed about this.

Reply:
Might keep as is -- my reasoning for liking having functions like this is it takes away cognitive load when debugging (as you're not presented with a lot of unneeded information)

Reply:
IMO / conversely I find such functions can cause more effort because you have to scroll to the very bottom to see what it's doing, before you can continue debugging from the top, especially when there is only 1x async call involved - I don't see benefit to creating a function purely to call another function

Reply:
removed